packages/create-app/templates/default-app/packages/backend/Dockerfile

```diff
@@ -1,11 +1,12 @@
 FROM node:12-buster
+ARG NPM_TOKEN
```

Comment:
Please note that this will persist the NPM token in the container. See the [ARG documentation in Docker](https://docs.docker.com/engine/reference/builder/#arg):

> Warning:
>
> It is not recommended to use build-time variables for passing secrets like github keys, user credentials etc. Build-time variable values are visible to any user of the image with the docker history command.
>
> Refer to the [“build images with BuildKit”](https://docs.docker.com/develop/develop-images/build_enhancements/#new-docker-build-secret-information) section to learn about secure ways to use secrets when building images.

It would be better to use the `--secret` option from BuildKit. But that is a bit more complex to use. An alternative is to use a `builder` container and copy all the files into the final one. But I am not sure if that really removes the secret from the image.

Reply:
Is this arg actually used?

Reply:
`.npmrc` included in this PR reads `NPM_TOKEN` env var during the build.

This allows us to build our backend docker image with plugins from our private NPM registry.

Reply:
I just realise `.npmrc` was no included in this PR. I'll switch it to `--secret`

Reply:
`yarn build-image --secret id=NPM_TOKEN,src=/NPM_TOKEN.txt`